pkg/kubelet/container/helpers.go

```diff
@@ -85,10 +86,38 @@ func ShouldContainerBeRestarted(container *v1.Container, pod *v1.Pod, podStatus
 			klog.V(4).Infof("Already successfully ran container %q of pod %q, do nothing", container.Name, format.Pod(pod))
 			return false
 		}
+		// Check restart count
+		if ExceedMaxRetries(pod, status.RestartCount) {
+			klog.V(5).Infof("Container %s restart count %d is greater than max, do nothing",
+				status.Name, status.RestartCount)
+			return false
+		}
+		klog.V(5).Infof("Container %s restart count is %d, will retry", status.Name, status.RestartCount)
 	}
 	return true
 }
 
+func getMaxRetries(pod *v1.Pod) int {
+	if pod.Annotations == nil {
+		return v1.MaxRetriesDefault
+	}
+	str, ok := pod.Annotations[v1.MaxRetriesAnnotation]
+	if !ok {
+		return v1.MaxRetriesDefault
+	}
+	if v, err := strconv.ParseInt(str, 10, 32); err != nil {
+		klog.Errorf("Failed to parse max retries(%s) for OnFailure policy: %v", str, err)
+		return v1.MaxRetriesDefault
+	} else {
```

Comment:
nit: the else is not necessary

Reply:
Fixing this will resolve the golint errors :)

Reply:
I need to use the value `v` in the return.

Reply:
you can remove the last if statement/block altogether and just use this as the default case if none of the other conditions are met